@here/olp-sdk-dataservice-read/lib/StatisticsClient.ts

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright (C) 2019 HERE Europe B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+import { CoverageApi } from "@here/olp-sdk-dataservice-api";
+import { ErrorHTTPResponse } from "./CatalogClientCommon";
+import { DataStoreContext } from "./DataStoreContext";
+import { DataStoreRequestBuilder } from "./DataStoreRequestBuilder";
+import { SummaryRequest } from "./SummaryRequest";
+
+/**
+ * A class that provides possibility to get Statistic Metadata and Data for Versioned layer
+ */
+export class StatisticsClient {
+    constructor(private readonly context: DataStoreContext) {
+        this.context = context;
+    }
+
+    /**
+     * Fetch and return layer summary from the Statistics service.
+     *
+     * @param summaryRequest Options which are needed to fetch summary.
+     * Includes @catalogHrn and @layerId
+     *
+     * @returns A promise with the layer summary.
+     */
+    async getSummary(summaryRequest: SummaryRequest): Promise<CoverageApi.LayerSummary> {
+        if (summaryRequest.catalogHrn === undefined) {
+            return Promise.reject(new Error(`No catalogHrn provided`));
+        }
+        if (summaryRequest.layerId === undefined) {
+            return Promise.reject(new Error(`No layerId provided`));
+        }
+        const coverageRequestBuilder = await this.getRequestBuilder(summaryRequest.catalogHrn)
+            .catch(error => error);
```

Comment:
.catch(error => error); Maybe should be returned Promise.reject from getSummary ?

Reply:
fixed